################################################################################
SHORT DESCRIPTION: 
################################################################################
Fix problem where time jumps forward in windows guest

################################################################################
LONG DESCRIPTION: 
################################################################################
Original comments:

 Fix problem where time jumps forward in windows guest

 Ensure that the current_tm field is updated before processing a request
 to modify time as this is used to calculate the delta between the current
 guest time and the new time.

The reproduction steps are not known but the patch is being kept for now since
it seems to fix a real issue.

################################################################################
CHANGELOG 
################################################################################

################################################################################
REMOVAL 
################################################################################
Possibly remove it.

################################################################################
UPSTREAM PLAN
################################################################################
No.

################################################################################
INTERNAL DEPENDENCIES 
################################################################################

################################################################################
PATCHES 
################################################################################
--- a/xen/arch/x86/hvm/rtc.c
+++ b/xen/arch/x86/hvm/rtc.c
@@ -578,6 +578,9 @@ static void rtc_set_time(RTCState *s)
       
     ASSERT(spin_is_locked(&s->lock));
 
+    /* Refresh current time */
+    *tm = gmtime(get_localtime(d));
+
     before = mktime(get_year(tm->tm_year), tm->tm_mon + 1, tm->tm_mday,
 		    tm->tm_hour, tm->tm_min, tm->tm_sec);
     
